addons/account/models/account_payment.py

```diff
@@ -702,6 +705,92 @@ def _compute_payment_receipt_title(self):
         """ To override in order to change the title displayed on the payment receipt report """
         self.payment_receipt_title = _('Payment Receipt')
 
+    @api.depends('partner_id', 'amount', 'date', 'payment_type')
+    def _compute_duplicate_move_ids(self):
+        """ Retrieve move ids of payments with same partner_id, amount and date as the
+            current payment, representing a credit or debit in the same account.
+        """
+        payment_to_duplicate_move = self._fetch_duplicate_reference()
+        for payment in self:
+            # Uses payment._origin.id to handle records in edition/existing records and 0 for new records
+            moves = payment_to_duplicate_move.get(payment._origin.id, self.env['account.move'])
+            # Remove moves with reconciled entries from the result
+            payment.duplicate_move_ids = moves - moves.filtered(lambda m: m.has_reconciled_entries)
```

Comment:
We can discuss it in person, but will leave a few thoughts here :) Maybe I'm having trouble understanding how reconciled works, but I couldn't make it work a 100% just with  `not dup_move_line.reconciled` 

For example:
![image](https://github.com/odoo/odoo/assets/47562677/60eeeb99-5798-4d49-a9e4-d27b45967fba)
Imagine move 45 is the payment we're interested in checking for duplicates. Move 64 (payment) and move 66 (bank statement) are reconciled. But since we match incoming payments based on whether they are a credit on the receivable account, for move 64 we use move line 147, which is not marked as reconciled. Just using `not dup_move_line.reconciled`  leads to the payment (move 64) still being queried.

I checked with Laura and the expected behavior is that both the payment and the statement line are reconciled, so neither should show as duplicate. For now, I added this condition to the query, instead of filtering the account.move: 
`AND dup_move_line.move_id NOT IN (
           SELECT move_id FROM account_move_line WHERE account_move_line.reconciled
       )`